sphere-models/src/main/java/io/sphere/sdk/products/messages/ProductPublishedMessage.java

```diff
@@ -0,0 +1,39 @@
+package io.sphere.sdk.products.messages;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import io.sphere.sdk.messages.GenericMessageImpl;
+import io.sphere.sdk.messages.MessageDerivateHint;
+import io.sphere.sdk.products.Product;
+import io.sphere.sdk.products.ProductProjection;
+import io.sphere.sdk.queries.PagedQueryResult;
+
+import java.time.ZonedDateTime;
+
+import static io.sphere.sdk.products.messages.MessagesPackage.*;
+
+@JsonDeserialize(as = ProductPublishedMessage.class)//important to override annotation in Message class
```

Comment:
The base interface has the same annotation for the default behavior of an untyped message. But the Jackson parser forces then all implementing classes to have also this annotation.